types/react-select/index.d.ts

```diff
@@ -17,9 +17,9 @@ export = ReactSelectClass;
 
 declare namespace ReactSelectClass {
     // Other components
-    class Creatable<TValue = OptionValues> extends React.Component<ReactCreatableSelectProps<TValue>> { }
-    class Async<TValue = OptionValues> extends React.Component<ReactAsyncSelectProps<TValue>> { }
-    class AsyncCreatable<TValue = OptionValues> extends React.Component<ReactAsyncSelectProps<TValue> & ReactCreatableSelectProps<TValue>> { }
+    class Creatable<TValue = OptionValues> extends React.Component<ReactCreatableSelectProps<TValue>, {}> { }
```

Comment:
[`React.Component`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/2e8fc9bb9820a7a03e96a9658544f2bf81d91e3b/types/react/index.d.ts#L274) already has a generic default on both parameters, so you don't need to pass the state type.

Reply:
Nice catch... I forgot about it.